### ISSUE TITLE
Comments on DCL implementation guide

### ISSUE THREAD

**Issue by phax** [ausdigital/ausdigital-dcl#8]
Hi there!
here are just a few things I stumbled upon when looking through the DCL implementation guide:
  * Figure 1: imho the arrow from DCP to DCL is missing (for initial registration).
  * In the paragraph below Figure 1 you say "The Digital Capability Locator is used by Access Points to enable the eInvoicing process." - alternatively you may state, that an AP can also directly use the DNS system, and therefore the DCL is never touched (and no single point of failure).
  * Chapter 5 defines `Schema` evern though it is never used - anyway an XSD with all the request and response types of the management API would be beneficiary
  * Bibliography entries 43, 44 and 45 can be removed
hth

**Comment by monkeypants** [comment]
Hey @matt-lewis-ato, not sure if you saw this feedback (ping).

> Figure 1: imho the arrow from DCP to DCL is missing (for initial registration).

Do you mean the DCP actually participates in the initial registration, or just that the initial registration makes reference to a DCP?

**Comment by phax** [comment]
In PEPPOL we have the following steps involved to get an SMP (DCP) registered at the SML (DCL)
  * Setup and configure SMP with certificates
  * Register SMP at SML which internally creates the "publisher" DNS entry (see our SMP spec). This is the entry which is referenced by all the "B-base32(md5(id))...." DNS entries and therefore you can change all participant URLs with a single change operation (the one on the publisher entry).
==> and therefore there is an arrow from SMP to SML in PEPPOL, and I thought your setup is (in this point) quite similar

**Comment by phax** [comment]
I think I mean the chapter `8.4.1 Update Digital Capability Publisher End-Point` - just for create and not for update :)

**Comment by monkeypants** [comment]
> and therefore you can change all participant URLs with a single change operation (the one on the publisher entry).

In our system, the DCP may update the DCL, but it's not the case that _only_ a DCP can do it. Any system can do it as long as:

 * They are a on OIDC relying party of an IDP
 * That IDP is trusted by the DCL (to an appropriate level of identity assurance)
 * The participant is a client of the IDP
 * The system has obtained a valid authorisation token from the IDP, with the authentication claim of the participant identifier, and the authorisation scope that lets them update business metadata on the participant's behalf.

So the DCP does not need certificates as such (except as an regular HTTPS endpoint). And the DCL may allow any system to update it (on behalf of participants), as long as they can prove (to the DCL's satisfaction) that they are authorised by the participant to do so.

Exactly the same authorisation mechanism is used to update DCPs and DCLs (see `ausdigital-idp` protocol). The only differences are the individual policies of the DCL/DCP operators (basically, choosing which IDP services to trust and how much to trust them).

The intent was to address issues we perceive in the certificate-based approach:

 * There is no appropriate certificate authority for this use-case
 * Unilaterally trusted DCPs present a large, porous attack surface with no market-mechanism for repair (perimeter-security is only as strong as the weakest provider, and the more providers there are the less secure it becomes).
 * Perimeter-security models don't scale well, because they require bureaucratic processes to ensure minimum-standard compliance ("trust the process" security, not "security by design")

In contrast to this, the consent-based authorisation model:

 * Is largely decentralised; the only central authority (which sits behind the DCL) is the statutory authority of the business identity scheme (The Australian Business Register, or some equivalent service operated by Commonwealth of Australia). There is no reason it couldn't support multiple identity schemes / jurisdictions, so long as each scheme is administered by someone who is responsible to the DCL of that scheme.
 * Encourages a trust market of Identity Providers, where poor security performance could be punished locally by network operators. Adding more providers makes the system more secure because it increases competition and reduces the blast-radius of compromised systems.
 * Removes the incentive and opportunity for cartel-like behavior in network operators.
 * Presents a low implementation barrier to developers with modern tooling.

Basically, we wanted to avoid designing a fallible bureaucratic process into the system (certificate authority), provide as much control and power to the participants as possible, and try to align the security objective with the market's self-interest.

**Comment by phax** [comment]
Thanks a lot for this very interesting summary. It seems like the similarities with PEPPOL are a bit less than I thought first :) So only the latter 3 (minor) issues are left. I will dive deeper into it

**Comment by monkeypants** [comment]
It's not simply another instanciation of the OASIS standards, but most of the components are fully backwards-compatible with OASIS counterparts. Most of the differences are concentrated around the IDP spec and the intra-network trust system.  In theory it should be fully interoperable with (send and receive documents) with PEPOL, however there is work ahead of us to verify/ensure that is true.

I think we need a page explaining similarities and differences with PEPOL.
